src/components/SelectedCardList.tsx

```diff
@@ -26,8 +26,11 @@ export const SelectedCardList: React.FC<Props> = ({
     setSelectedCards(filteredCards)
   }
 
-  const handleReset = () => {
-    setSelectedCards([])
+  const handleUndo = () => {
+    const filteredCards = selectedCards.filter(
+      (_, index) => index !== selectedCards.length - 1
+    )
```

Comment:
[slice](https://developer.mozilla.org/ja/docs/Web/JavaScript/Reference/Global_Objects/Array/slice)のほうが直感的な気がする

```suggestion
    const filteredCards = selectedCards.slice(0, selectedCards.length - 1)
```

Array.prototype.pop でいいと思ってたけど副作用あるから使えないのか。

Reply:
修正しました！